tests/PHPStan/Analyser/nsrt/preg_match_shapes.php

```diff
@@ -570,3 +570,33 @@ function (string $s): void {
 		assertType("array{0: string, 1: non-empty-string, 2?: numeric-string, 3?: 'x'}", $matches);
 	}
 };
+
+function (string $s): void {
+	if (preg_match('/Price: ([a-z])/i', $s, $matches)) {
+		assertType("array{string, non-empty-string}", $matches);
+	}
+};
+
+function (string $s): void {
+	if (preg_match('/Price: ([0-9])/i', $s, $matches)) {
+		assertType("array{string, numeric-string}", $matches);
+	}
+};
+
+function (string $s): void {
+	if (preg_match('/Price: ([xXa])/i', $s, $matches)) {
+		assertType("array{string, 'a'|'X'|'x'}", $matches);
+	}
+};
+
+function (string $s): void {
+	if (preg_match('/Price: (ba[rz])/i', $s, $matches)) {
+		assertType("array{string, 'bar'|'baz'}", $matches);
```

Comment:
`i` modifier must be honored, ie. `'baR'` must be present in the expected type

Reply:
At that point I'd fall back to not have literal strings at all. Too many combinations -> performance problems

Reply:
Yes, had the same thing in mind. Will fix. 

Thanks